diff --git a/fc-service-core/src/main/java/eu/gaiax/difs/fc/core/service/verification/impl/VerificationServiceImpl.java b/fc-service-core/src/main/java/eu/gaiax/difs/fc/core/service/verification/impl/VerificationServiceImpl.java
index 6daa65be..71e4dee5 100644
--- a/fc-service-core/src/main/java/eu/gaiax/difs/fc/core/service/verification/impl/VerificationServiceImpl.java
+++ b/fc-service-core/src/main/java/eu/gaiax/difs/fc/core/service/verification/impl/VerificationServiceImpl.java
@@ -46,6 +46,7 @@ import org.apache.jena.vocabulary.RDF;
 import org.bouncycastle.jce.provider.BouncyCastleProvider;
 
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Configuration;
 import org.springframework.stereotype.Component;
 import org.topbraid.shacl.validation.ValidationUtil;
 import org.topbraid.shacl.vocabulary.SH;
@@ -56,6 +57,8 @@ import java.net.URI;
 import java.net.URL;
 import java.nio.charset.StandardCharsets;
 import java.security.GeneralSecurityException;
+import java.security.KeyManagementException;
+import java.security.NoSuchAlgorithmException;
 import java.security.Security;
 import java.security.cert.CertificateException;
 import java.security.cert.CertificateFactory;
@@ -70,6 +73,8 @@ import org.apache.jena.riot.system.stream.StreamManager;
 import org.springframework.beans.factory.annotation.Qualifier;
 import org.springframework.beans.factory.annotation.Value;
 
+import javax.net.ssl.*;
+
 import static org.apache.jena.rdf.model.ResourceFactory.*;
 
 
@@ -618,14 +623,11 @@ public class VerificationServiceImpl implements VerificationService {
     String[] uri_parts = uri.getSchemeSpecificPart().split(":");
     String did_json;
     if (uri_parts[0].equals("web")) {
-      String[] _parts = uri_parts[1].split("#");
-      URL url;
-      if (_parts.length == 1) {
-        url = new URL("https://" + _parts[0] + "/.well-known/did.json");
-      } else {
-        url = new URL("https://" + _parts[0] + "/.well-known/did.json#" + _parts[1]);
-      }
-      log.debug("readDIDFromURI; requesting DIDDocument from: {}", url.toString());
+      String didWeb = uri.toString()
+              .replace("did:web:", "") // remove did type prefix
+              .replaceFirst("#.*", ""); // remove verification method reference
+      URL url = new URL(getDidDocumentUri(didWeb));
+      log.debug("readDIDFromURI; requesting DIDDocument from: {}", url);
       InputStream stream = url.openStream();
       did_json = IOUtils.toString(stream, StandardCharsets.UTF_8);
     } else {
@@ -636,6 +638,26 @@ public class VerificationServiceImpl implements VerificationService {
     return result;
   }
 
+  private static String getDidDocumentUri(String didWeb) {
+    boolean containsSubpath = didWeb.contains(":");
+    StringBuilder didDocumentUriBuilder = new StringBuilder();
+    didDocumentUriBuilder.append(didWeb
+            .replace(":", "/") // Replace ":" with "/" in the method specific identifier to
+            // obtain the fully qualified domain name and optional path.
+            .replace("%3A", ":")); // If the domain contains a port percent decode the colon.
+
+    // Generate an HTTPS URL to the expected location of the DID document by prepending https://.
+    didDocumentUriBuilder.insert(0, "https://");
+    if (!containsSubpath) {
+      // If no path has been specified in the URL, append /.well-known.
+      didDocumentUriBuilder.append("/.well-known");
+    }
+    // Append /did.json to complete the URL.
+    didDocumentUriBuilder.append("/did.json");
+
+    return didDocumentUriBuilder.toString();
+  }
+
   private Map<String, Object> extractRelevantVerificationMethod(List<Map<String, Object>> methods, URI verificationMethodURI) {
     //TODO wait for answer https://gitlab.com/gaia-x/lab/compliance/gx-compliance/-/issues/22
     log.debug("extractRelevantVerificationMethod; methods: {}, uri: {}", methods, verificationMethodURI);
@@ -695,10 +717,6 @@ public class VerificationServiceImpl implements VerificationService {
     //Second, extract required information
     Instant exp = relevant.getNotAfter().toInstant();
 
-    if (!checkTrustAnchor(uri)) {
-      throw new VerificationException("Signatures error; The trust anchor is not set in the registry. URI: " + uri);
-    }
-
     log.debug("hasPEMTrustAnchorAndIsNotDeprecated.exit; returning: {}", exp);
     return exp;
   }
@@ -971,3 +989,49 @@ public class VerificationServiceImpl implements VerificationService {
   }
 
 }
+
+@Configuration
+@Slf4j
+class SSLConfig {
+  /**
+   * Disables the SSL certificate checking for new instances of {@link HttpsURLConnection} This has been created to
+   * aid testing on a local box, not for use on production.
+   */
+
+  public SSLConfig(@Value("${federated-catalogue.ignore-ssl:false}") boolean ignoreSsl) {
+    if (ignoreSsl) {
+      disableSSLCertificateChecking();
+      log.info("Disabling SSL verification for web requests.");
+    }
+  }
+
+  private static void disableSSLCertificateChecking() {
+    TrustManager[] trustAllCerts = new TrustManager[] { new X509TrustManager() {
+      public X509Certificate[] getAcceptedIssuers() {
+        return null;
+      }
+
+      @Override
+      public void checkClientTrusted(X509Certificate[] arg0, String arg1) throws CertificateException {
+        // Not implemented
+      }
+
+      @Override
+      public void checkServerTrusted(X509Certificate[] arg0, String arg1) throws CertificateException {
+        // Not implemented
+      }
+    } };
+
+    try {
+      SSLContext sc = SSLContext.getInstance("TLS");
+
+      sc.init(null, trustAllCerts, new java.security.SecureRandom());
+
+      HttpsURLConnection.setDefaultSSLSocketFactory(sc.getSocketFactory());
+    } catch (KeyManagementException e) {
+      e.printStackTrace();
+    } catch (NoSuchAlgorithmException e) {
+      e.printStackTrace();
+    }
+  }
+}
