diff --git a/fc-service-core/src/main/java/eu/xfsc/fc/core/service/schemastore/SchemaStoreImpl.java b/fc-service-core/src/main/java/eu/xfsc/fc/core/service/schemastore/SchemaStoreImpl.java
index 05ad1b35..4d643f66 100644
--- a/fc-service-core/src/main/java/eu/xfsc/fc/core/service/schemastore/SchemaStoreImpl.java
+++ b/fc-service-core/src/main/java/eu/xfsc/fc/core/service/schemastore/SchemaStoreImpl.java
@@ -69,6 +69,10 @@ public class SchemaStoreImpl implements SchemaStore {
   @Override
   public int initializeDefaultSchemas() {
     log.debug("initializeDefaultSchemas.enter");
+
+    // Always start fresh
+    clear();
+
     int count = 0;
     int found = dao.getSchemaCount();
     if (found == 0) {
